### ISSUE TITLE
Benchmark.t fails due time prediction failure with TEST_JOBS=2 + make test_harness

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13963]
Migrated from [rt.perl.org#122202](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=122202) (status was 'resolved')

Searchable as RT122202$

**Comment by p5pRT** [comment]
### From @kentfredric

perl&#8203;: 5\.18\.2


It doesn't make much sense to me to have a test that relies on CPU time being predictable\.

https://gist.github.com/kentfredric/264d2b9a0f169f72c148

\_\_
TESTFILE=harness LD\_LIBRARY\_PATH=/tmp/portage/dev\\\-lang\\\-perl\\\-5\.18\.2/work/perl\\\-5\.18\.2 \./runtests choose
No saved state\, selection will be empty
base/cond\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\. ok   

\.\.\.

\.\./lib/AnyDBM\_File\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\. ok
===\( 243983;1819 16532/? 10/196 \)===================================
\# Failed test 'is 50\.6378802747792 within 0\.4 of estimate \(159\.34\)'
\# at \.\./lib/Benchmark\.t line 93\.
\# '2\.14665620155039'
\# \<=
\# '0\.4'
\# in\_threesecs = 1603
\# in\_threesecs\_adj = 478\.031809145129
\# cpu3 = 10\.06
\# sys3 = 0\.91
\# estimate = 159\.34
\# in\_onesec = 516
\# in\_onesec\_adj = 50\.6378802747792
\# cpu1 = 10\.19
\# sys1 = 1\.17
\.\./lib/Benchmark\.t \.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\.\. Dubious\, test returned 1 \(wstat 256\, 0x100\)
Failed 1/196 subtests 

There's a reference or two to this being a problem with TEST\_JOBS\, and a suggested patch\, its just not clear that that patch ever got merged&#8203;:

http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2010/02/msg156265\.html

And there's similar failures from jenkins I dug up&#8203;: 

http&#8203;://www\.nntp\.perl\.org/group/perl\.perl5\.porters/2012/09/msg193348\.html 

Overall\, it just seems unwise to have this test being something that can fail to start with\, CPU time is notoriously unpredictable\. 

For instance\, if you're running close to capacity and don't have ideal cooling \(ie&#8203;: laptops\)\, thermal throttle can kick in winding CPU clock speed back to a mere 33% of standard performance\, and that could be momentary\, or somewhat lengthy\, and it can enter a state of hysteresis\, bursting between high performance and throttled at every 5 seconds\.

\( This computer in question is not usually subject to that specific hysteresis issue\, but I have another one that has that problem routinely \)

**Comment by p5pRT** [comment]
### From @iabyn

On Tue\, Jul 01\, 2014 at 02&#8203;:43&#8203;:08AM \-0700\, Kent Fredric wrote&#8203;:
> \# New Ticket Created by  Kent Fredric 
> \# Please include the string&#8203;:  \[perl \#122202\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=122202 >
> 
> 
> perl&#8203;: 5\.18\.2
> 
> 
> It doesn't make much sense to me to have a test that relies on CPU time being predictable\.
> 
> https://gist.github.com/kentfredric/264d2b9a0f169f72c148

Indeed\, which is why we removed that test for 5\.20 :\-\)

\-\- 
Red sky at night \- gerroff my land\!
Red sky at morning \- gerroff my land\!
&nbsp;    \-\- old farmers' sayings \#14

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @kentfredric

Oh sweet\. =\)\.

I for some reason had the misfortune of triggering this scenario on every subsequent build/test run I did\, so figured I'd change the value of TEST\_JOBS to 4 instead of 2 to change my luck as to \*which\* tests would get run in parallel\, and it seemed to have been sufficient\.

I guess if its fixed in 5\.20 then we can close this bug\, and it can serve as a "how to avoid this problem" marker if anyone else stumbles upon it building older perls\.

**Comment by p5pRT** [comment]
### From @jkeenan

On Tue Jul 01 14&#8203;:39&#8203;:12 2014\, kentfredric wrote&#8203;:

> 
> I guess if its fixed in 5\.20 then we can close this bug\, and it can
> serve as a "how to avoid this problem" marker if anyone else stumbles
> upon it building older perls\.

Closing per recommendation of original poster\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
